forage-android/src/main/java/com/joinforage/forage/android/core/services/vault/CapturePaymentRepository.kt

```diff
@@ -38,37 +37,24 @@ internal class CapturePaymentRepository(
             else -> return response
         }
 
-        val vaultResponse = when (
-            val response = vaultSubmitter.submit(
-                params = VaultSubmitterParams(
-                    encryptionKeys = encryptionKeys,
-                    idempotencyKey = paymentRef,
-                    merchantId = merchantId,
-                    path = AbstractVaultSubmitter.capturePaymentPath(paymentRef),
-                    paymentMethod = paymentMethod,
-                    userAction = UserAction.CAPTURE,
-                    sessionToken = sessionToken
-                )
+        val response = vaultSubmitter.submit(
+            params = VaultSubmitterParams(
+                encryptionKeys = encryptionKeys,
+                idempotencyKey = paymentRef,
+                merchantId = merchantId,
+                path = AbstractVaultSubmitter.capturePaymentPath(paymentRef),
+                paymentMethod = paymentMethod,
+                userAction = UserAction.CAPTURE,
+                sessionToken = sessionToken
             )
-        ) {
-            is ForageApiResponse.Success -> Message.ModelMapper.from(response.data)
-            else -> return response
-        }
-
-        val pollingResponse = pollingService.execute(
-            contentId = vaultResponse.contentId,
-            operationDescription = "payment capture of Payment $paymentRef"
         )
-        if (pollingResponse is ForageApiResponse.Failure) {
-            return pollingResponse
-        }
 
-        return when (val paymentResponse = paymentService.getPayment(paymentRef)) {
+        return when (response) {
             is ForageApiResponse.Success -> {
                 logger.i("[HTTP] Received updated Payment $paymentRef for Payment $paymentRef")
```

Comment:
Can we update the log statement now that we don't do a GET for the `Payment`? Also lol that we include the payment ref twice in this string

Reply:
great callout about updating the log statement! Ha! I didn't see that we double-captured `paymentRef` until just now :P

Reply:
Your feedback showed that we can actually drop `when` blocks in both the `CapturePaymentRepository` class and the `CheckBalanceRepository` class. Now that I've done that, I'll mark this thread as resolved